command/debug.go

```diff
@@ -551,28 +615,51 @@ func (c *DebugCommand) collectConsul(dir, consul string) error {
 
 // collectVault calls the Vault API directly to collect data
 func (c *DebugCommand) collectVault(dir, vault string) error {
-	if vault == "" {
+	addr := c.vault.addr(vault)
+	if addr == "" {
 		return nil
 	}
 
-	token := c.vaultToken
-	if token == "" {
-		os.Getenv("VAULT_TOKEN")
-	}
-
-	client := http.Client{
-		Timeout: 2 * time.Second,
-	}
+	client := api.DefaultHttpClient()
+	api.ConfigureTLS(client, c.vault.tls)
 
-	req, _ := http.NewRequest("GET", vault+"/sys/health", nil)
-	req.Header.Add("X-Vault-Token", token)
+	req, _ := http.NewRequest("GET", addr+"/sys/health", nil)
+	req.Header.Add("X-Vault-Token", c.vault.token())
 	req.Header.Add("User-Agent", userAgent)
 	resp, err := client.Do(req)
 	c.writeBody(dir, "vault-sys-health.json", resp, err)
 
 	return nil
 }
 
+func (c *external) addr(addr string) string {
+	if c.addrVal == "" {
+		return addr
+	}
+	if c.ssl && c.addrVal[0:5] != "https" {
+		return "https" + c.addrVal[4:]
+	}
```

Comment:
This makes few implicit assumptions about https vs http relationship, and validity of input.  To avoid panics, or unexpected modifications if user passed invalid protocol, maybe this would be better:

```suggestion
	if c.ssl && strings.HasPrefix(c.addrVal, "http:") {
		return "https:" + c.addrVal[5:]
	}
```

Also, looks like consul doesn't require `CONSUL_HTTP_ADDR` to contain the protocol: https://www.consul.io/docs/commands#consul_http_addr

Reply:
oh, thanks, good catch!